<?xml version="1.0"?>
<!--
Software License Agreement (BSD)

\file      husky_playpen.launch
\authors   Paul Bovbel <pbovbel@clearpathrobotics.com
\copyright Copyright (c) 2015, Clearpath Robotics, Inc., All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that
the following conditions are met:
 * Redistributions of source code must retain the above copyright notice, this list of conditions and the
   following disclaimer.
 * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
   following disclaimer in the documentation and/or other materials provided with the distribution.
 * Neither the name of Clearpath Robotics nor the names of its contributors may be used to endorse or promote
   products derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WAR-
RANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, IN-
DIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>

  <arg name="mav_name" default="hummingbird"/>
  <arg name="robot_namespace" default="husky"/>
  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="log_file" default="$(arg mav_name)" />
  <arg name="laser_enabled" default="$(optenv HUSKY_LASER_LMS1XX_ENABLED 1)"/>
  <arg name="realsense_enabled" default="$(optenv HUSKY_REALSENSE_ENABLED 0)"/>
  <arg name="start_idx" default="-1"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find rps_planning)/worlds/basic.world" />
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <rosparam command="load" file="$(find rps_planning)/config/UAV_UGV.yaml" />
  <node name="age_server_node" pkg="rps_planning" type="age_server.py" output="screen"/>

  <include file="$(find rps_planning)/launch/spawn_team.launch">
    <arg name="ugv_idx" value="0"/>      
    <arg name="UGV_name" value="0"/>
     
    <arg name="mav_name" value="$(arg mav_name)"/>
    <arg name="enable_logging" value="$(arg enable_logging)" />
    <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
    <arg name="log_file" value="$(arg log_file)" />
    <arg name="laser_enabled" value="$(arg laser_enabled)"/>
    <arg name="realsense_enabled" value="$(arg realsense_enabled)"/>
    <arg name="robot_namespace" value="$(arg robot_namespace)0"/>
    <!-- UGV initial pose-->
    <arg name="UGV_x" value="2.5"/> 
    <arg name="UGV_y" value="5.5"/>
    <arg name="UGV_z" value="0.0"/>
    <arg name="UGV_yaw" value="0.0"/>
    <!-- UAV global index-->
    <arg name="uav0" value="0"/>
    <arg name="uav1" value="1"/>
    <arg name="uav2" value="2"/>
    <arg name="uav3" value="3"/>

    <!-- if set to i, in gazebo the UGV will start monitoring the i+1 partition-->
    <arg name="start_idx" value="-1"/>   
  </include>


  

</launch>
